src/main/java/seedu/address/logic/parser/AddressBookParser.java

```diff
@@ -87,7 +87,7 @@ public Command parseCommand(String userInput) throws ParseException {
                 return new FindCommandParser().parse(personType, arguments);
 
```

Comment:
I'm not sure, but I think the intended behavior was that delete commands are no longer used as `delete -pa 1` but just `delete 1`? I think it would be better to remove the unused tag than have it be redundant. 
In that case, the Delete commands can be moved to the `else if (matcherBasic.matches())` branch below. I probably should have left some comments, but the first branch is for commands that use `-pa` or `-sp` tags, and the second branch is for commands that don't. Would be great if you could add those comments too!

Reply:
Ok, I assume this would make the DeleteCommandParser class unnecessary as well, since it is implementing the Parser interface which mandates a personType argument (deleting DeleteCommandParser would address the other comment below as well).

Reply:
Hm, I didn't think about this. I feel the `DeleteCommandParser` class should still exist since we still need to parse the index. 
I guess we'll have to keep the PersonType parameter in the method then as you've done, but we'll have to pass it as a null when you restructure the `AddresBookParser` since we won't actually have a PersonType to pass for a Delete command (this would be a temporary fix).
One possible better fix would be eventually creating two sub `Parser` interfaces, one that parses commands with `PersonType` and one that parses commands without, but I also think more thought can be put into this. We could do the temporary fix for now, and open an issue to refactor this part of the code in the future.

Reply:
Sorry, I saw this comment a bit too late but I actually ended up implementing ParserBasic and ParserComplex classes similar to what you've suggested here to solve the issue!